api/database/migrations/2023_10_17_183718_add_foreign_key_constraints.php

```diff
@@ -0,0 +1,104 @@
+<?php
+
+use Illuminate\Database\Migrations\Migration;
+use Illuminate\Database\Schema\Blueprint;
+use Illuminate\Support\Facades\Schema;
+
+return new class extends Migration
+{
+    /**
+     * Run the migrations.
+     */
+    public function up(): void
+    {
+        Schema::table('applicant_filter_classification', function (Blueprint $table) {
```

Comment:
I'm not sure I agree with all these delete cascades.  If a team is deleted then delete all their pools? :scream: I think we need to think these through.

When there's a very tight "composition" relationship that exists then it makes sense.  I'd agree with user-permission.  If the user is deleted it makes sense for their permissions to also disappear.  And some of the pivots.  If an applicant_filter is deleted it makes sense to remove its pivots to classifications.  But if a classification is deleted I'm not sure I want all the search filters that depend on it to silently disappear.

This seems pretty out of scope for this PR.  Unless you want to go through these carefully one-by-one I'd suggest not including the delete cascades at all unless it really breaks a delete use case that's in use in the app.  We can always add them later if we need to.

Reply:
Hm, okay. I thought for things like pivots a delete cascade was a safe thing. But if it isn't I can wind it back a bit. 
I got rid of some in https://github.com/GCTC-NTGC/gc-digital-talent/pull/8240/commits/e08d3ba68de7973c02dc151029c9d1312dfbbf19

Basically only kept deletion cascading in cases of references to the users table. We do have user deletion so I thought that was the one cascade operation best to keep around.